src/Microsoft.VisualStudio.Services.Agent/Constants.cs

```diff
@@ -228,5 +229,17 @@ public static class Task
                 public static readonly string DisplayName = "task.displayname";
             }
         }
+
+        public class Release
+        {
+            public class Path
+            {
+                public static readonly string WorkingDirectory = "r";
```

Comment:
Please do not add yet another "working directory" . System.DefaultWorkingDirectory (Build.WorkingDirectory in build, Release.WorkingDirectory in release) is the working directory. It looks like the folder you're adding here is neither. This is very confusing.

I don't understand the concern. The computation logic changes *very little* with the addition of an `r` prefix. See following:
```csharp
Regex regex = new Regex(@"^[0-9]*$");
 +                var dirs = Directory.GetDirectories(releaseDirectory);
 +                var integerFolderNames = dirs.Select(Path.GetFileName).Where(name => regex.IsMatch(name));
```
versus:
```csharp
Regex regex = new Regex(@"^r[0-9]*$");
 +                var dirs = Directory.GetDirectories(releaseDirectory);
 +                var integerFolderNames = dirs.Select(Path.GetFileName).Where(name => regex.IsMatch(name)).Select(x => x.Substring(1));
```

I think it's worth it to not introduce yet another working directory concept that is not really the working directory.